lib/handler/RedirectHandler.js

```diff
@@ -24,6 +24,15 @@ class BodyAsyncIterable {
 }
 
 class RedirectHandler {
+  static buildDispatch (dispatcher, maxRedirections) {
+    if (maxRedirections != null && (!Number.isInteger(maxRedirections) || maxRedirections < 0)) {
+      throw new InvalidArgumentError('maxRedirections must be a positive number')
+    }
+
+    const dispatch = dispatcher.dispatch.bind(dispatcher)
+    return (opts, originalHandler) => dispatch(opts, new RedirectHandler(dispatch, maxRedirections, opts, originalHandler))
+  }
```

Comment:
Yea... this is ugly but we can figure out somehting better in a follow up

Reply:
Yeah, didn't like it either; it sorts out the compatibility issue for now but still does not feel something ergonomic for end-users. I'll open a follow up for Redirect and the rest of handlers 👍